inc/utility/class-handlestransfers.php

```diff
@@ -0,0 +1,347 @@
+		$upload = $this->handleUpload();
+
+		if ( ! $upload ) {
+			return;
+		}
+
+		$items = $this->getImportableItems( $upload );
+
+		unlink( $upload['file'] );
+
+		if ( empty( $items ) ) {
+			return;
+		}
+
+		$this->handleImport( $items );
+
+		$_SESSION['pb_notices'][] = __( 'Successfully imported.', 'pressbooks' );
+	}
+
+	/**
+	 * Generates and download a JSON file with all the selected resources.
+	 *
+	 * @param $ids
+	 * @return void
+	 */
+	public function exportTaxonomyList( $ids ) {
+		$items = $this->getExportableItems( $ids );
+
+		if ( empty( $items ) ) {
+			return;
+		}
+
+		$this->downloadJson( wp_json_encode( $items ) );
+	}
+
+	/**
+	 * Return a list of taxonomy terms that should be exported.
+	 *
+	 * @param $ids
+	 * @return array
+	 */
+	public function getExportableItems( $ids ) {
+		$items = [];
+		$fields = $this->getTransferableFields();
+
+		foreach ( $ids as $id ) {
+			$term = get_term( $id, self::TAXONOMY );
+			$term_meta = get_term_meta( $term->term_id );
+
+			$item = [
+				'name' => $term->name,
+				'slug' => $term->slug,
+			];
+
+			foreach ( $fields as $field ) {
+				$value = $term_meta[ $field ] ?? [];
+
+				$item[ $field ] = $value[0] ?? '';
+			}
+
+			$items[] = $item;
+		}
+
+		return $items;
+	}
+
+	/**
+	 * Download the content as a JSON file.
+	 *
+	 * @param string $content
+	 * @return void
+	 */
+	public function downloadJson( $content ) {
+		$filename = self::TAXONOMY . '-list-' . time() . '.json';
+
+		$file = create_tmp_file();
+
+		put_contents( $file, $content );
+
+		force_download( $file, false, $filename );
+
+		if ( ! defined( 'WP_TESTS_MULTISITE' ) ) {
+			exit;
+		}
+	}
+
+	/**
+	 * Handles file upload before importing contributors.
+	 *
+	 * @return array|false
+	 */
+	public function handleUpload() {
+		if ( empty( $_FILES['import_file']['name'] ) ) {
+			return false;
+		}
+
+		if ( $_FILES['import_file']['type'] !== 'application/json' ) {
+			$_SESSION['pb_errors'][] = __( 'Sorry, this file type is not permitted for security reasons.' );
+
+			return false;
+		}
+
+		$upload = wp_handle_upload( $_FILES['import_file'], [
+			'test_type' => false,
+			'action' => self::TAXONOMY . '-import',
+		] );
+
+		if ( ! empty( $upload['error'] ) ) {
+			$_SESSION['pb_errors'][] = $upload['error'];
+
+			return false;
+		}
+
+		return $upload;
+	}
+
+	/**
+	 * Returns the list of contributors to import
+	 *
+	 * @param array $upload
+	 * @return array
+	 */
+	public function getImportableItems( $upload ) {
+		$items = [];
+		$invalid_rows = [];
+
+		foreach ( json_decode( file_get_contents( $upload['file'] ) ) as $key => $item ) {
+			$item = (array) $item;
+
+			if ( ! isset( $item['name'], $item['slug'] ) ) {
+				$invalid_rows[] = $key + 1;
+
+				continue;
+			}
+
+			$items[] = $item;
+		}
+
+		if ( ! empty( $invalid_rows ) ) {
+			$message = __( 'One or more rows were not imported because they have missing information. Please, check rows %s', 'pressbooks' );
```

Comment:
This is a temporary message. 

If the json structure does not contain `name` or `slug`, the contributor won't be imported and we could let the user know. Once functionality is in place we could think of a proper message here.

Reply:
Makes sense, @SteelWagstaff could you confirm this message?

Reply:
I'd prefer `The following rows were not imported because they are missing the name or slug: (%s)`